inference-engine/ie_bridges/java/sample/face_detection_sample_async/Main.java

```diff
@@ -0,0 +1,241 @@
+
+                    String outputName =  new ArrayList<String>(net.getOutputsInfo().keySet()).get(0);
+    
+                    ExecutableNetwork executableNetwork = core.LoadNetwork(net, device);
+                    
+                    Vector<InferRequest> inferRequests = new Vector<InferRequest>();
+                    Vector<Boolean> asyncInferIsFree = new Vector<Boolean>(inferRequestsSize);
+
+                    for (int i = 0; i < inferRequestsSize; i++) {
+                        inferRequests.add(executableNetwork.CreateInferRequest());
+                        asyncInferIsFree.add(true);
+                    }
+
+                    Queue<Integer> startedRequestsIds = new LinkedList<Integer>();
+    
+                    boolean isRunning = true;
+
+                    while (isRunning) {
+                        while (!startedRequestsIds.isEmpty()) {
+                            processInferRequets(outputName, startedRequestsIds, asyncInferIsFree, inferRequests);
+                            break;
+                        }
+
+                        for (int i = 0; i < inferRequestsSize; i++) {
+                            if (!asyncInferIsFree.get(i))
+                                continue;
+
+                            Mat frame = framesQueue.poll(0, TimeUnit.SECONDS);
+
+                            if (frame == null) {
+                                isRunning = false;
+                                break;
+                            }
+
+                            InferRequest request = inferRequests.get(i);
+        
+                            asyncInferIsFree.setElementAt(false, i);
+                            processedFramesQueue.add(frame);  // predictionsQueue is used in rendering
+
+                            Blob imgBlob = imageToBlob(frame);
+                            request.SetBlob(inputName, imgBlob); 
+
+                            startedRequestsIds.add(i);
+                            request.StartAsync();
+                        }
+                    }
+                    while (!startedRequestsIds.isEmpty()) {
+                        processInferRequets(outputName, startedRequestsIds, asyncInferIsFree, inferRequests);
+                    }
+                    return;
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                    
+                    for (Thread t : Thread.getAllStackTraces().keySet())
+                        if (t.getState()==Thread.State.RUNNABLE) 
+                            t.interrupt(); 
+                }
+            }
+        });
+
+        captureThread.start();
+        inferThread.start();
+
+        VideoWriter out = null;
+        int i = 0;
+
+        TickMeter tm = new TickMeter();
+        tm.start();
+
+        try {
+            while (true) {
+                float[] detection = detectionOutput.poll(waitingTime, TimeUnit.SECONDS);
+                if(detection != null) {                   
```

Comment:
Let's have the following termination criteria:
1. Capturing thread is terminated by by `false` from `cap.read(frame)` - done
2. Processing thread is terminated by timeout from capturing thread
3. Main visualization should finish in case of terminated processing thread or by empty results queue:
```java
while (inferThread.isAlive() || !detectionOutput.empty())
````